### PR TITLE
Good friends don't let their friends commit without linting or formatting

### PR DESCRIPTION
**Pull request by jlyon87**
# IMPORTANT 
This installs node dev dependencies. Accepting this PR means the project requires Node and NPM to run. Merging this PR means you'll need to `npm install` for the changes to take effect. 

---
## Changes
This PR adds a pre-commit hook that executes prettier. Prettier will "prettify" only the files staged for commit, and will automatically re-`git add` those files to the commit. The effect is a near silent change in place to prettify files just before they are committed. 

## Info
Prettier Rules continue to respect the .prettierrc. Additional tooling may be added to the pre-commit hook such as ESLint. 

Configure the hook by changing `ling-staged` in the package.json.

## Recommendation
To take full advantage, I recommend adding ESLint and setting up lint and format on save in your editor settings, leveraging the ESLint and Prettier plugins for your editor of choice.

## Sources
[lint-staged](https://www.npmjs.com/package/lint-staged)
[husky](https://www.npmjs.com/package/husky)
[prettier](https://prettier.io/docs/en/index.html)

### CODE REVIEW

**Comment by tsalb** [comment]
LOL yes, it's time to get this repo up to public contribution shape.

I got away with it for a while since I was the only contributor but as you said, it's time to formalize a few things.

**Comment by tsalb** [comment]
FWIW, the eventual plan is to tease out which dependencies (inspired from [lwc-recipes](https://github.com/trailheadapps/lwc-recipes/blob/master/package.json) are needed in this repo.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "lwc-utils",
+  "version": "1.0.0",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/jlyon/lwc-utils.git"
```
Comment (jamessimone) [inline package.json]: I would fix this up to point to tsalb's repo

File: package.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  ],
+  "author": "jlyon",
+  "bugs": {
+    "url": "https://github.com/jlyon87/lwc-utils/issues"
+  },
+  "homepage": "https://github.com/jlyon/lwc-utils#readme",
+  "devDependencies": {
+    "husky": "^4.3.0",
+    "lint-staged": "^10.4.0",
+    "prettier": "^2.1.2"
```
Comment (jamessimone) [inline package.json]: missing prettier-plugin-apex

**Comment by tsalb** [comment]
@jamessimone @jlyon87 

Ah, I think we simultaneously commit + commented. See [this commit](https://github.com/tsalb/lwc-utils/pull/31/commits/b69c04f3c763adbaf01a2644deb9c1ed00fa0346) that re-aligns the package.json

**Comment by jamessimone** [comment]
@tsalb 🤦‍♂️ classic concurrency problem. Nicely done.

**Comment by jlyon87** [comment]
I'm getting 546 errors running lint on utils-core with: `npx eslint --quiet --fix utils-core/main/default/lwc`

Ideally we should be able to run `eslint --quiet fix` against the whole project at least once before we add it to the commit hook.

Until then, the `lint` script I added to package.json is just a shortcut to prettier.

**Comment by jlyon87** [comment]
Additionally, I worked out that we could consolidate .eslintrc to the project root by adding the relative path from lwc/.eslintrc to the root/.eslintrc in the lwc/ local's `extends` property.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# CONTRIBUTING
```
Comment (jlyon87) [inline CONTRIBUTING.md:1]: Do we want to provide Contributing Guidelines?

Comment (tsalb) [reply CONTRIBUTING.md:1]: IMO, not in this PR, we should focus on getting project.json barebones running.

**Comment by tsalb** [comment]
Btw, I tried using this config on my private repo (works like a charm!) but I ran into some nuances where i needed to `.prettierignore` most `.xml` files since the way husky is set up it yells at you and doesn't let you commit (without `no-verify`).

Given that, is there any downside to adding `*.xml` to the `.prettierignore` file in context of this specific repo?

**Comment by jlyon87** [comment]
I don't see why not - I do not have the prettier default memorized, there
may be some odd bits that we bump into.

**Comment by tsalb** [comment]
@jlyon87 BTW, I've been holding on this PR because I wasn't sure you were done with it (based on our discord chat).

LMK if you had any additional plans for it otherwise I can start looking into the impact of merging this to the repo!

**Comment by jlyon87** [comment]
Yep, this is good to go. 🚀 ship it 🚀 !!

**Pull request merged**